sat/cli/bootprep/input/image.py

```diff
@@ -639,14 +639,42 @@ def product_version(self):
         return self.image_data['base']['product'].get('version')
 
     @property
-    def base_description(self):
-        """str: a human-readable description of the base we are starting from"""
+    def filter_prefix(self):
+        """str or None: the filter prefix to use for filtering images or recipes from the product"""
+        return self.image_data['base']['product'].get('filter', {}).get('prefix')
+
+    @property
+    def filter_func(self):
+        """Callable: a function that returns True if the given name matches the filter, False otherwise"""
+        if not self.filter_prefix:
+            return lambda x: True
+        else:
+            return lambda x: x.startswith(self.filter_prefix)
```

Comment:
I'm a little unsure why this needs to be a property returning a function, could it be simplified with the following?
```suggestion
    def filter_func(self, name):
        """Check if the name matches the filter"""
        if not self.filter_prefix:
            return True
        return x.startswith(self.filter_prefix)
```
At callsites of the higher-order function, the usage _should_ be exactly the same, but there's one less level of abstraction involved.

Reply:
Yep, looks good. I guess it was just late when I implemented this 🤷

Reply:
I've also added a test for when the filter prefix is the empty string, which I believe this was handling incorrectly before, and now it is handling correctly (empty string prefix matches all).

Reply:
Actually, it was handling it fine since `name.startswith(self.filter_prefix)` evaluates to True just as `not self.filter_prefix` when `self.filter_prefix` is the empty string.